### PR TITLE
Bug fix #1884

### PR DESCRIPTION
**Pull request by hoand-vn**
tested + reviewed : Okie

### CODE REVIEW

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     )),
                     new Assert\Regex(array(
                         'pattern' => '/^\d+$/u',
-                        'message' => 'form.type.numeric.invalid',
+                        'message' => $app->trans('form.type.numeric.invalid'),
```
Comment (t-nagahashi) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: If you fix another problem exclude this issue, please commnet on this pullrequest.

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php [hunk 2]
```diff
@@ -69,6 +69,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             if ($data['id'] && empty($data['name'])) {
                 $form['name']->addError(new FormError($app->trans('This value should not be blank.')));
             }
+
+            if ($data['name'] && $data['id'] == '') {
```
Comment (t-nagahashi) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: Please use empty method( dont use "==" in this case.)
Do you have reason of using "=="?

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php (hunk 1 above)
Comment (ryo-endo) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: この程度の内容なら、１つのPRで合わせて修正するのも許容範囲かなと思います。
ただ、Issueに書かれていること以外を修正した場合は、レビュー時に把握したいので連絡欲しいですね。

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php [hunk 3]
```diff
@@ -69,6 +69,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             if ($data['id'] && empty($data['name'])) {
```
Comment (t-nagahashi) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: Please fix this.
from "empty" to "==".

Comment (ryo-endo) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: ref http://tech.aainc.co.jp/archives/2535

Comment (t-nagahashi) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: please use "==".

Please confirm this test case.
when you input "0" to name and id, this case is OK.

Comment (lqdung-lockon) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: yes, I will fix it.

**Comment by hoand-vn** [comment]
fixed condition. Please check it again.

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php [hunk 4]
```diff
@@ -53,15 +55,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
             ))
             ->add('name', 'text', array(
                 'required' => false,
             ))
-            ;
+        ->addEventListener(FormEvents::POST_SUBMIT, function (FormEvent $event) use ($app) {
+            $form = $event->getForm();
+            $data = $form->getData();
+            if (strlen($data['id']) && strlen($data['name']) == 0) {
+                $form['name']->addError(new FormError($app->trans('This value should not be blank.')));
```
Comment (ryo-endo) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: `$app->trans` is not need. (please, delete.)

ref:
dump($app->trans('This value should not be blank.'));
=> `"This value should not be blank."` (its meaningless)

File: src/Eccube/Form/Type/Admin/MasterdataDataType.php [hunk 5]
```diff
@@ -53,15 +55,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+            }
+
+            if (strlen($data['name']) && strlen($data['id']) == 0) {
+                $form['id']->addError(new FormError($app->trans('This value should not be blank.')));
```
Comment (ryo-endo) [inline src/Eccube/Form/Type/Admin/MasterdataDataType.php]: same L70

**Comment by ryo-endo** [comment]
ok.
please, pr to master after the test fixed.

**Pull request closed without merging**